pkg/validate/mezvalidator.go

```diff
@@ -0,0 +1,359 @@
+package validate
+
+import (
+	"encoding/json"
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/Eyevinn/mp4ff/mp4"
+)
+
+// MezValidationResult holds the results of MP4 validation
+type MezValidationResult struct {
+	Valid                    bool
+	SampleCount              int
+	UniformSampleDuration    bool
+	ExpectedSampleDuration   uint32
+	ActualSampleDurations    []uint32
+	MFHDSequenceNumbers      []uint32
+	TFDTBaseTimes            []uint64
+	ExpectedBaseTime         uint64
+	TotalDuration            uint64
+	Timescale                uint32
+	FileSize                 int64
+	Issues                   []string
+}
+
+// ValidateMez validates an fMP4 mezzanine segment file using mp4ff library
+func ValidateMez(filename string) (*MezValidationResult, error) {
+	file, err := os.Open(filename)
+	if err != nil {
+		return nil, fmt.Errorf("failed to open file: %v", err)
+	}
+	defer func() {
+		if closeErr := file.Close(); closeErr != nil {
+			fmt.Printf("Warning: failed to close file: %v\n", closeErr)
+		}
+	}()
+
+	// Get file size
+	fileInfo, err := file.Stat()
+	if err != nil {
+		return nil, fmt.Errorf("failed to get file info: %v", err)
+	}
+	fileSize := fileInfo.Size()
+
+	// Parse the MP4 file using mp4ff
+	parsedMP4, err := mp4.DecodeFile(file)
+	if err != nil {
+		return nil, fmt.Errorf("failed to parse MP4 file: %v", err)
+	}
+
+	result := &MezValidationResult{
+		Valid:                 true,
+		Issues:                []string{},
+		ActualSampleDurations: []uint32{},
+		MFHDSequenceNumbers:   []uint32{},
+		TFDTBaseTimes:         []uint64{},
+		FileSize:              fileSize,
+	}
+
+	// Extract timescale from moov if present
+	if parsedMP4.Moov != nil && len(parsedMP4.Moov.Traks) > 0 {
+		result.Timescale = parsedMP4.Moov.Traks[0].Mdia.Mdhd.Timescale
+	}
+
+	// Analyze fragments
+	err = analyzeFragments(parsedMP4, result)
+	if err != nil {
+		return nil, fmt.Errorf("failed to analyze fragments: %v", err)
+	}
+
+	// Validate the extracted information
+	validateResults(result)
+
+	return result, nil
+}
+
+// analyzeFragments extracts information from movie fragments
+func analyzeFragments(parsedMP4 *mp4.File, result *MezValidationResult) error {
```

Comment:
Is there a way that this function could actually return an error? In this case, there are no errors in the fragment analyzing code.

Do you think there are reasonable cases where the fragments could be so badly formed that it causes an error, or should that have errored at some point in `mp4.DecodeFile`?

Reply:
Good point. We could add more tests here like checking that all normalt boxes are present and report an error if not, but it is not a likely situation, so it is probably better to remove the error for now.